backend-rs/src/utils.rs

```diff
@@ -0,0 +1,425 @@
+    pub fn fetch_all(device: &Device) -> Result<Vec<CameraMode>, Box<dyn Error>> {
+        let mut configs: Vec<CameraMode> = Vec::new();
+
+        let discrete_iter = Capture::enum_framesizes(device, FourCC::new(b"MJPG"))?
+            .into_iter()
+            .flat_map(|framesize| framesize.size.to_discrete());
+
+        for discrete_frame_size in discrete_iter {
+            for interval in Capture::enum_frameintervals(
+                device,
+                FourCC::new(b"MJPG"),
+                discrete_frame_size.width,
+                discrete_frame_size.height,
+            )? {
+                let fraction = match interval.interval {
+                    FrameIntervalEnum::Discrete(fraction) => fraction,
+                    FrameIntervalEnum::Stepwise(Stepwise { min, .. }) => min,
+                };
+
+                configs.push(CameraMode {
+                    width: discrete_frame_size.width,
+                    height: discrete_frame_size.height,
+                    frame_interval: fraction,
+                });
+            }
+        }
+
+        Ok(configs)
+    }
+}
+
+// Handles communication between CameraThreadHandle(s) and WebRTC clients.
+pub struct WebcamManager {
+    camera_handles: Arc<Mutex<Vec<CameraThreadHandle>>>,
+    rtc_api: API,
+}
+
+impl WebcamManager {
+    pub fn new() -> Result<Self, Box<dyn Error>> {
+        let mut media_engine = MediaEngine::default();
+        media_engine.register_default_codecs()?;
+
+        let mut registry = Registry::new();
+        registry =
+            interceptor_registry::register_default_interceptors(registry, &mut media_engine)?;
+
+        let rtc_api = APIBuilder::new()
+            .with_media_engine(media_engine)
+            .with_interceptor_registry(registry)
+            .build();
+
+        Ok(Self {
+            camera_handles: Arc::new(Mutex::new(Vec::new())),
+            rtc_api,
+        })
+    }
+
+    pub fn camera_handles(&self) -> Arc<Mutex<Vec<CameraThreadHandle>>> {
+        self.camera_handles.clone()
+    }
+
+    // IMPORTANT: Everything in here runs within the tokio runtime!
+    pub async fn add_client(
+        &self,
+        camera_path: String,
+        rtc_offer: RTCSessionDescription,
+    ) -> Result<RTCSessionDescription, Box<dyn Error>> {
+        let peer_connection = self
+            .rtc_api
+            .new_peer_connection(RTCConfiguration::default())
+            .await?;
+        let peer_connection = Arc::new(peer_connection);
+        let peer_disconnected = Arc::new(AtomicBool::new(false));
+        let notify_tx = Arc::new(Notify::new());
+        let notify_rx = notify_tx.clone();
+
+        let video_track = Arc::new(TrackLocalStaticSample::new(
+            RTCRtpCodecCapability {
+                mime_type: MIME_TYPE_H264.to_owned(),
+                ..Default::default()
+            },
+            "video".to_owned(),
+            "webrtc".to_owned(),
```

Comment:
If you create multiple streams under the same IDs, what happens? I'm actually not sure, but I'm concerned that it may result in unexpected behavior. Please make sure the constant IDs are intentional.

Remember: it's very important that folks can understand this later on. May and June are going to be chock-full of debugging, so we want to get it out of the way now! :D